sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -732,18 +722,21 @@ object DataSource extends Logging {
   def lookupDataSourceV2(provider: String, conf: SQLConf): Option[TableProvider] = {
```

Comment:
Should we remove the changes in the `lookupDataSource` method above? Since a Python data source can only be a V2 source.
```scala
else if (isUserDefinedDataSource) {
  classOf[PythonTableProvider]
} 
```
and 
```scala
case head :: Nil =>
  // there is exactly one registered alias
  // TODO(SPARK-45600): should be session-based.
  val isUserDefinedDataSource = SparkSession.getActiveSession.exists(
    _.sessionState.dataSourceManager.dataSourceExists(provider))
  // The source can be successfully loaded as either a V1 or a V2 data source.
  // Check if it is also a user-defined data source.
  if (isUserDefinedDataSource) {
    throw QueryCompilationErrors.foundMultipleDataSources(provider)
  }
  head.getClass
```

Reply:
It depends on whether we will call `lookupDataSource` to detect python data sources. Seems no?

Reply:
`lookupDataSource` detects both V1 and V2 classes. `lookupDataSourceV2` just creates the instance from V2 classes.

Reply:
Yes, but technically python data source is not v1 or v2. So it seems ok if `lookupDataSource` doesn't return python data sources.

Reply:
But when we extend the Python Data Source features, we will use v2 code path, for example, `ExternalCommandRunner`. Same as `DataStreamReader.loadInternal` code path.